Sources/Sentry/SentryCrashScopeObserver.m

```diff
@@ -0,0 +1,131 @@
+#import <Foundation/Foundation.h>
+#import <NSData+Sentry.h>
+#import <SentryCrashJSONCodec.h>
+#import <SentryCrashJSONCodecObjC.h>
+#import <SentryCrashScopeObserver.h>
+#import <SentryLog.h>
+#import <SentryScopeSyncC.h>
+#import <SentryUser.h>
+
+@implementation SentryCrashScopeObserver
+
+- (void)setUser:(nullable SentryUser *)user
+{
+    NSDictionary *serialized = @{ @"user" : [user serialize] };
+
+    NSData *json = [self getBytes:serialized];
+    if (json == nil) {
+        return;
+    }
+
+    sentryscopesync_setUserJSON([json bytes]);
```

Comment:
I believe calling `setUser(null)` should behave as `removeUser`, so we should not bail out but rather calling `sentryscopesync_setUserJSON(null)` or having a method to remove the user

Reply:
this applies to the other methods if it makes sense

Reply:
Yes, I agree. Good point thanks.